keel-core/src/main/kotlin/com/netflix/spinnaker/keel/services/ResourceStatusService.kt

```diff
@@ -93,6 +95,12 @@ class ResourceStatusService(
     return first() is ResourceCreated
   }
 
+  private fun List<ResourceHistoryEvent>.isWaiting(): Boolean {
+    // we expect to have only two events, but we will accept several different "unresolvable" events
+    // in order to be less brittle and show the user this status instead of an error
+    return size < 5 && all { it is ResourceCreated || it is ResourceCheckUnresolvable }
```

Comment:
The `5` seems arbitrary here. What's the thinking there?

Reply:
+1. Maybe make it a constant with a self-explanatory name?

Reply:
Totally arbitrary. I feel like if the environment is actually new it'll only have a couple of events. I wanted to make sure this new status only shows up in cases where it is definitely new. So, I chose 5. Does that make sense / seem reasonable?

Reply:
@robfletcher @luispollo I was hoping my comment of `// we expect to have only two events, but we will accept several different "unresolvable" events` would convey why I chose 5, is there a way I could make this comment more clear?